opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -85,6 +88,21 @@ def update(self, value: metrics_api.ValueT):
         self.last_update_timestamp = time_ns()
         self.aggregator.update(value)
 
+    def release(self):
+        self.decrease_ref_count()
```

Comment:
Could you avoid all this by making `handles` a [`WeakValueDictionary`](https://docs.python.org/3/library/weakref.html#weakref.WeakValueDictionary)?

Reply:
I was experiment with it and I think it's not possible. `handles` **must** rename in memory until they are collected, if I use `WeakValueDictionary` those would be deleted before collecting, missing their updates.